### PR TITLE
Increasing Spotbugs level and fixing Spotbugs issues

### PR DESCRIPTION
**Pull request by aneveux**
Hello folks :wave:

I'm currently following the [Improve a Plugin documentation](https://www.jenkins.io/doc/developer/tutorial-improve/) for
Jenkins plugins modernization, and noticed we could [add more Spotbugs checks](https://www.jenkins.io/doc/developer/tutorial-improve/add-more-spotbugs-checks/) in this plugin.

So here's a PR following the documentation recommendation by increasing the Spotbugs checks, as well as fixing the
reported issues.

Let me know if that sounds interesting to you or if any other modifications are needed.

Thanks a lot!

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira // I do not have a Jira issue for that. Should I create one?
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfig.java [hunk 1]
```diff
@@ -130,6 +134,8 @@ public S3BlobStoreConfig getConfiguration() {
         }
     }
 
+    @SuppressFBWarnings(value = "MC_OVERRIDABLE_METHOD_CALL_IN_CONSTRUCTOR",
+            justification = "Calling load in the constructor is the typical way to manage GlobalConfiguration objects.")
```
Comment (aneveux) [inline src/main/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfig.java:138]: I guess as an alternative I could also make `S3BlobStoreConfig` implement `PersistentDescriptor`. Would you prefer that?

File: src/main/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfig.java [hunk 2]
```diff
@@ -101,18 +105,18 @@ public class S3BlobStoreConfig extends AbstractAwsGlobalConfiguration {
...
+
+        @SuppressFBWarnings(value = "SE_TRANSIENT_FIELD_NOT_RESTORED", justification = "This transient field is only modified from the class constructor.")
         private transient S3BlobStoreConfig config;
 
-        S3BlobStoreTester(String container, String prefix, boolean useHttp,
-            boolean useTransferAcceleration, boolean usePathStyleUrl,
-            boolean disableSessionToken, String customEndpoint,
-            String customSigningRegion) {
+        S3BlobStoreTester(String container, String prefix, boolean useHttp, boolean useTransferAcceleration, boolean usePathStyleUrl,
+                          boolean disableSessionToken, String customEndpoint, String customSigningRegion) {
```
Comment (jglick) [inline src/main/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfig.java:119]: Please avoid gratuitous formatting changes.

File: src/main/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfig.java (hunk 1 above)
Comment (jglick) [reply src/main/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfig.java:138]: No, https://github.com/jenkinsci/aws-global-configuration-plugin/blob/106dc1d8d86ee77da24c3bcf6f30bcf5959446e8/src/main/java/io/jenkins/plugins/aws/global_configuration/AbstractAwsGlobalConfiguration.java#L36 is correct as it stands.

The warning could be addressed just by making this class `final`.

File: src/test/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfigTest.java [hunk 3]
```diff
@@ -7,6 +7,9 @@
 import java.util.logging.Logger;
 import jenkins.model.ArtifactManagerConfiguration;
 import static org.junit.Assert.*;
+
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.core.IsInstanceOf;
```
Comment (jglick) [inline src/test/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfigTest.java:12]: Conventionally import these from `Matchers`.

File: src/main/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfig.java (hunk 1 above)
Comment (aneveux) [reply src/main/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfig.java]: Oh, that's right. Thanks a lot!

I updated it.

File: src/main/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfig.java (hunk 2 above)
Comment (aneveux) [reply src/main/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfig.java]: Reverted, sorry for that.

File: src/test/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfigTest.java (hunk 3 above)
Comment (aneveux) [reply src/test/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfigTest.java]: I'm not sure I understood your comment, do you mean I shouldn't use hamcrest and use something else instead?

File: src/test/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfigTest.java [hunk 4]
```diff
@@ -6,10 +6,11 @@
 import io.jenkins.plugins.artifact_manager_jclouds.JCloudsArtifactManagerFactory;
 import java.util.logging.Logger;
 import jenkins.model.ArtifactManagerConfiguration;
+
+import static org.hamcrest.core.IsInstanceOf.instanceOf;
```
Comment (jglick) [inline src/test/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfigTest.java:10]: Fine though we conventionally import these things via `Matchers`.

File: src/test/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfigTest.java (hunk 3 above)
Comment (jglick) [reply src/test/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfigTest.java]: ⇒ https://github.com/jenkinsci/artifact-manager-s3-plugin/pull/339#discussion_r1062568555

File: src/test/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfigTest.java (hunk 4 above)
Comment (jglick) [reply src/test/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfigTest.java:10]: https://hamcrest.org/JavaHamcrest/javadoc/2.1/org/hamcrest/Matchers.html#instanceOf-java.lang.Class- in this example

Comment (aneveux) [reply src/test/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfigTest.java]: Oh ok :) I fixed that, thanks a lot for the reference :)

File: src/main/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfig.java [hunk 5]
```diff
@@ -129,7 +135,7 @@ public S3BlobStoreConfig getConfiguration() {
             return config;
         }
     }
-
+
```
[the remark this answers is not in the document]
Comment (aneveux) [reply src/main/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfig.java]: Again an automated formatting I missed :'( Sorry about that

File: src/test/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfigTest.java [hunk 6]
```diff
@@ -1,15 +1,22 @@
...
-import jenkins.model.ArtifactManagerConfiguration;
-import static org.junit.Assert.*;
+
 import org.junit.Rule;
 import org.junit.Test;
 import org.jvnet.hudson.test.JenkinsRule;
+import io.jenkins.plugins.artifact_manager_jclouds.BlobStoreProvider;
+import io.jenkins.plugins.artifact_manager_jclouds.JCloudsArtifactManagerFactory;
+
+import hudson.model.Failure;
```
Comment (jglick) [inline src/test/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfigTest.java:11]: Unfortunate. Guess I need to set up Spotless to enforce simple lexicographical `import` sorting.

**Review by jglick — APPROVED** [review]
Thanks!

**Pull request merged**